src/com/xilinx/rapidwright/interchange/DeviceResourcesVerifier.java

```diff
@@ -224,8 +224,8 @@ private static boolean verifySiteType(Device device, DeviceResources.Device.Read
             if( (isInput != (dir == Direction.INPUT)) || (isOutput != (dir == Direction.OUTPUT)) ){
                 throw new RuntimeException("ERROR: Mismatch on site pin direction");
             }
-            Integer siteWireIndex = site.getSiteWireIndex(pinName);
-            expect(siteWireIndex == null ? -1 : siteWireIndex, spReader.getSitewire());
+            String tileWireName = site.getTileWireNameFromPinName(pinName);

```

Comment:
See my other comment from DeviceResourcesWriter.java:

Corresponding changes to match suggested changes would be:
```
            Integer siteWireIndex = site.getSiteWireIndex(pinName);
            String siteWireName = siteWireIndex == null ? null : site.getSiteWireName(siteWireIndex);
            int siteWireNameIdx = spReader.getSitewire();
            expect(siteWireName == null ? null : siteWireName, siteWireNameIdx == -1 ? null : allStrings.get(siteWireNameIdx));
```
Also, because the siteWire associated with a site pin can be null, we have to accommodate that scenario.  The `expect()` method would also need to be modified by checking for null equality:

```
    private static boolean expect(String gold, String test) {
        if(gold == null && test == null) return true;
        //...
```

Reply:
Fixed